File: Dante1/Module1.05/Zadanie5.08/main.c
#include <stdio.h>
#include <ctype.h>
#include <string.h>

int main()
{
    char input_text[100];
    char reversed_text[100];
    char final_text[1000];

    printf("Enter your text: ");
    scanf("%99[^\n]s", input_text);

    if(strlen(input_text) >= 99)
    {
        printf("Input is too long");
        return 2;
    }

    size_t our_input_text_size = strlen(input_text);

    int i;
    for(i = 0; i < (int)our_input_text_size; i++)
    {
        reversed_text[i] = input_text[our_input_text_size - i - 1];
    }
    reversed_text[i] = '\0';

    if(isdigit(reversed_text[0]))
    {
        printf("Incorrect input");
        return 1;
    }

    for(i = 0; i < (int)our_input_text_size; i++)
    {
        if(isdigit(reversed_text[i]) && isdigit(reversed_text[i + 1]))
        {
            printf("Incorrect input");
            return 1;
        }
    }

    int j = 0;
    for(i = 0; i < (int)our_input_text_size; i++)
    {
        int number = 1;
        if(!isdigit(reversed_text[i]))
        {
            final_text[j] = reversed_text[i];
            j++;
        }
        else
        {
            while(number != reversed_text[i] - 48)
            {
                final_text[j] = reversed_text[i - 1];
                number++;
                j++;
            }
        }
    }
    final_text[j] = '\0';

    //printf("\n%i", (int)strlen(final_text));
    if(j > 200)
    {
        printf("Output is too long");
        return 3;
    }

    printf("\n%s",final_text);

    return 0;
}

File: Dante1/Module1.05/Zadanie5.08/main.rs
fn main() {
    let mut input: String = String::new();

    println!("Enter: ");
    std::io::stdin().read_line(&mut input).expect("Error");
    let input_text = &input;

    let size_of_input_text: usize = input_text.len();
    if size_of_input_text >= 99 {
        panic!("Input is too long");
    }

    let mut reserved_text: Vec<char> = vec![];
    for i in 0..size_of_input_text {
        reserved_text.push(input_text.chars().nth(size_of_input_text - i - 1).expect("popa"));
    }

    if reserved_text[0].is_digit(10) {
        panic!("Incorrect input");
    }

    for i in 0..size_of_input_text - 1 {
        if reserved_text[i].is_digit(10) && reserved_text[i + 1].is_digit(10) {
            panic!("Incorrect input");
        }
    }

    let mut final_text: Vec<char> = vec![];
    let mut j: usize = 0;
    for i in 0..size_of_input_text {
        let mut number: u8 = 1;
        if !reserved_text[i].is_digit(10) {
            final_text.push(reserved_text[i]);
            j += 1;
        }
        else {
            while number != reserved_text[i] as u8 {
                final_text.push(reserved_text[i - 1]);
                number += 1;
                j += 1;
            }
        }
    }

    if j > 200 {
        panic!("Output is too long");
    }

    println!("{:?}", final_text);
}
